google/cloud/spanner/read_partition.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/read_partition.h"
+#include <google/spanner/v1/spanner.pb.h>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+ReadPartition::ReadPartition(std::string transaction_id, std::string session_id,
+                             std::string partition_token,
+                             std::string table_name, KeySet key_set,
+                             std::vector<std::string> column_names,
+                             ReadOptions read_options) {
+  proto_.set_session(std::move(session_id));
+  proto_.mutable_transaction()->set_id(std::move(transaction_id));
+  proto_.set_table(std::move(table_name));
+  if (!read_options.index_name.empty()) {
+    proto_.set_index(std::move(read_options.index_name));
+  }
+  for (auto& column : column_names) {
+    *proto_.mutable_columns()->Add() = std::move(column);
+  }
+  *proto_.mutable_key_set() = internal::ToProto(std::move(key_set));
+  if (read_options.limit != 0) {
```

Comment:
It may be better to omit this check and always call `proto_.set_limit()` with whatever `read_options.limit` is set to. That way this class isn't applying any semantic meaning to these fields, it's just copying them through.

I suggest the same thing with `read_options.index_name` above.